Flaky `TestConvertClassicHistogramsToNHCB`

### What did you do?

Ran `scrape` unit tests multiple times in a loop (just looped using bash instead of using `count` argument.
```
go test -count=1 -race -run '^TestConvertClassicHistogramsToNHCB$' ./scrape
```

### What did you expect to see?

Consistent test behaviour.

### What did you see instead? Under which circumstances?
Sometimes the output is:

```

--- FAIL: TestConvertClassicHistogramsToNHCB (3.42s)
    scrape_test.go:4592: 
                Error Trace:    /home/buda/code/prometheus/scrape/scrape_test.go:4592
                                                        /home/buda/code/prometheus/scrape/scrape_test.go:4768
                Error:          Not equal: 
                                expected: 1
                                actual  : NaN
                Test:           TestConvertClassicHistogramsToNHCB
    --- FAIL: TestConvertClassicHistogramsToNHCB/convert_without_scrape_with_protobuf,_in_different_order (0.32s)
        testing.go:1679: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
```
Or:
```
--- FAIL: TestConvertClassicHistogramsToNHCB (0.34s)
    scrape_test.go:4592: 
                Error Trace:    /home/buda/code/prometheus/scrape/scrape_test.go:4592
                                                        /home/buda/code/prometheus/scrape/scrape_test.go:4768
                Error:          Not equal: 
                                expected: 1
                                actual  : NaN
                Test:           TestConvertClassicHistogramsToNHCB
    --- FAIL: TestConvertClassicHistogramsToNHCB/neither_scrape_nor_convert_with_protobuf (0.32s)
        testing.go:1679: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
2025/06/03 22:32:14 http: TLS handshake error from 127.0.0.1:52098: remote error: tls: bad certificate
FAIL
```


### System information

```
~$ go version
go version go1.24.3 linux/amd64
```

### Prometheus version

Executed against `main`, currently https://github.com/prometheus/prometheus/commit/40a27ce87591487239d07e5f47010f114ee5bea2


### Prometheus configuration file

```yaml

```

### Alertmanager version

```text

```

### Alertmanager configuration file

```yaml

```

### Logs

```text

```

--- a/scrape/scrape_test.go
+++ b/scrape/scrape_test.go
@@ -4695,7 +4695,7 @@ metric: <
 					SampleLimit:                    100,
 					Scheme:                         "http",
 					ScrapeInterval:                 model.Duration(50 * time.Millisecond),
-					ScrapeTimeout:                  model.Duration(25 * time.Millisecond),
+					ScrapeTimeout:                  model.Duration(49 * time.Millisecond),
 					AlwaysScrapeClassicHistograms:  tc.alwaysScrapeClassicHistograms,
 					ConvertClassicHistogramsToNHCB: tc.convertClassicHistToNHCB,
 					MetricNameValidationScheme:     config.UTF8ValidationConfig,
@@ -4747,7 +4747,7 @@ metric: <
 						Targets: []model.LabelSet{{model.AddressLabel: model.LabelValue(testURL.Host)}},
 					},
 				})
-				require.Len(t, sp.ActiveTargets(), 1)
+				require.Eventually(t, func() bool { return len(sp.ActiveTargets()) == 1 }, 5*time.Second, 50*time.Millisecond)
 
 				select {
 				case <-time.After(5 * time.Second):

